src/producer.ts

```diff
@@ -23,53 +22,55 @@ export class Producer {
         receiver.onMessage(message);
       }
     }
-    receiver.onMessage({ type: 'voiceEnd' });
   }
 
   private getMessages(node: Node): EasyScoreMessage[] {
     const messages: EasyScoreMessage[] = [];
     const nodeElem = node as Element;
     switch (node.nodeName) {
       case 'measure':
-        return Array.from(node.childNodes).flatMap(this.getMessages);
-      case 'attributes':
-        const children = node.childNodes;
-        for (let i = 0; i < children.length; i++) {
-          const childElem = children[i] as Element;
-          switch (children[i].nodeName) {
-            case 'clef':
-              const sign = childElem.getElementsByTagName('sign').item(0)?.textContent ?? '';
-              messages.push({ type: 'clef', clef: sign });
-              break;
-            case 'time':
-              const beats = childElem.getElementsByTagName('beats').item(0)?.textContent ?? '';
-              const beatType = childElem.getElementsByTagName('beat-type').item(0)?.textContent ?? '';
-              messages.push({ type: 'timeSignature', top: beats, bottom: beatType });
-              break;
-            default:
-              console.log(`unprocessed node, got: ${children[i]}`);
-          }
+        const width = nodeElem.getAttribute('width') ?? '';
+        const staves = nodeElem.getElementsByTagName('staves').item(0)?.textContent ?? '1';
+        const sign: string[] = [];
+        for (let i = 0; i < parseInt(staves); i++) {
+          sign.push(nodeElem.getElementsByTagName('sign').item(i)?.textContent ?? '');
         }
+        const beats = nodeElem.getElementsByTagName('beats').item(0)?.textContent ?? '';
+        const beatType = nodeElem.getElementsByTagName('beat-type').item(0)?.textContent ?? '';
+        messages.push({ msgType: 'measureStart', width, staves, clef: sign, time: `${beats}/${beatType}` });
+        messages.push(...Array.from(node.childNodes).flatMap(this.getMessages));
```

Comment:
nit: This was my fault, I think it needs to be `this.getMessages.bind(this)` to not lose context.

Reply:
Done